x-pack/plugin/eql/src/main/antlr/EqlBase.g4

```diff
@@ -21,43 +18,52 @@ singleExpression
 statement
     : query (PIPE pipe)*
     ;
- 
+
 query
     : sequence
     | join
-    | condition
+    | eventQuery
+    ;
+withParams
+    : WITH namedParam (COMMA namedParam)*
     ;
-    
 sequence
-    : SEQUENCE (by=joinKeys)? (span)?
-      match+
-      (UNTIL match)?
+    : SEQUENCE  (by=joinKeys withParams? | withParams by=joinKeys?)?
```

Comment:
Are there any other params/keys outside `span`? I couldn't find any in the docs.

Reply:
The position of `withParams` is confusing - it can be optional but only trailing however if it's leading is mandatory.
Why not mandate only one form and reduce ambiguity.

Reply:
Part of my motivation is dropping `maxspan` as a keyword so it can still be used as an identifier. I haven't added other toggleable features for seqeunces, but there were some parameters that I had in mind, like controlling the timing (maxspan, maxpause, etc), or other functionality like overlaps.

There is a named parameter for changing the behavior of a _term_ within the sequence. `fork=true`, is used to progress a sequence with copies instead of moves, so you can have multiple sequences with the same prefix. 

I don't see any added ambiguity to the grammar, since both rules have leading keywords `by` or `where`.
In general, having one form does make sense, but at this point I want to maintain as much compatibility as possible.

Reply:
>  I don't see any added ambiguity to the grammar, since both rules have leading keywords by or where.

It's ambiguity in intent as there are 2 different ways to do the exact same thing.
I agree that compatibility is important however it's worth understanding the impact of this change, for example how many queries are of type A and of type B? This should also give an indication of the de-facto choice for queries.

Such exceptions add up fast and add technical debt that gets harder to remove going forward (case in point, the current situation).